django_freeradius/api/views.py

```diff
@@ -16,27 +18,47 @@
 User = get_user_model()
 
 
-@api_view(['POST'])
-def authorize(request):
-    username = request.data.get('username')
-    password = request.data.get('password')
-    try:
-        user = User.objects.get(username=username, is_active=True)
-    except User.DoesNotExist:
-        user = None
-    if user and user.check_password(password):
-        return Response({'control:Auth-Type': 'Accept'}, status=200)
-    return Response({'control:Auth-Type': 'Reject'}, status=401)
+class TokenAuthMixin(object):
+    def token_auth(self, request, *args, **kwargs):
+        if not API_TOKEN:
+            return True
+        if request.META.get('HTTP_AUTHORIZATION', False):
+            headers = request.META.get('HTTP_AUTHORIZATION').split(',')
+            for header in headers:
+                token = header.split(' ')[1]
+                if token == API_TOKEN:
+                    return True
+        return False
+
+
+class AuthorizeView(TokenAuthMixin, APIView):
```

Comment:
Good to see this being converted to a class based view, but I don't understand.. why do we want to reply with the radius terminology?
Someone is trying to use our API even though we haven't authorized them, we can just cut them off, we don't care if freeradius will complain because of some error, we can just document this like: "if you don't configure freeradius to use the token you will get this error...".

I believe you should create a custom DRF authentication class which returns an anonymous user if successful or raises an exception if unsuccessful: http://www.django-rest-framework.org/api-guide/authentication/#custom-authentication.
You may also need to set the permission attribute as well.

Reply:
@nemesisdesign Yeah I just left kept it for testing purposes. We could change the reply here.

Custom authentication sounds good actually. Will modify accordingly.